aten/src/ATen/native/cuda/UniqueCub.cu

```diff
@@ -0,0 +1,201 @@
+    inverse_indices = at::empty({num_inp}, options);
+    Tensor inv_loc = consecutive ? at::empty({num_inp}, options.dtype(kInt))
+                                 : inverse_indices;
+    int* inv_loc_ptr = static_cast<int*>(inv_loc.data_ptr());
+    const dim3 block =
+        dim3(std::min(static_cast<int64_t>(cuda::getApplyBlock().x), num_inp));
+    dim3 grid;
+    int curDevice = -1;
+    cudaGetDevice(&curDevice);
+    cuda::getApplyGrid(num_inp, grid, curDevice);
+    adjacent_difference_kernel<scalar_t>
+        <<<grid, block, 0, stream>>>(num_inp, data, inv_loc_ptr);
+    C10_CUDA_KERNEL_LAUNCH_CHECK();
+
+    Tensor inv_loc_out =
+        consecutive ? inverse_indices : at::empty({num_inp}, options);
+    CUB_WRAPPER(
+        cub::DeviceScan::InclusiveSum,
+        inv_loc_ptr,
+        inv_loc_out.data_ptr<int64_t>(),
+        num_inp,
+        stream);
+
+    if (!consecutive) {
+      scatter_kernel<<<grid, block, 0, stream>>>(
+          num_inp,
+          inv_loc_out.data_ptr<int64_t>(),
+          sorted_indices.data_ptr<int64_t>(),
+          inverse_indices.data_ptr<int64_t>());
+      C10_CUDA_KERNEL_LAUNCH_CHECK();
+    }
+  }
+
+  // unique and count
+  Tensor data_out = at::empty({num_inp}, sorted.options());
+  Tensor counts = at::empty({0}, options);
+  Tensor length = at::empty({1}, options);
+  int64_t num_out;
+  if (!return_counts) {
+    CUB_WRAPPER(
+        cub::DeviceSelect::Unique,
+        data,
+        data_out.data_ptr<scalar_t>(),
+        length.data_ptr<int64_t>(),
+        num_inp,
+        stream);
+    num_out = length.item<int64_t>();
+  } else {
+    counts.resize_(num_inp);
+    CUB_WRAPPER(
+        cub::DeviceRunLengthEncode::Encode,
+        data,
+        data_out.data_ptr<scalar_t>(),
+        counts.data_ptr<int64_t>(),
+        length.data_ptr<int64_t>(),
+        num_inp,
+        stream);
+    num_out = length.item<int64_t>();
+    counts.resize_(num_out);
+  }
+
+  return std::tuple<Tensor, Tensor, Tensor, int64_t>(
+      data_out, inverse_indices, counts, num_out);
+}
+
+} // namespace
+
+template <typename scalar_t>
+std::tuple<Tensor, Tensor, Tensor> unique_cuda_template(
+    const Tensor& self,
+    const bool consecutive,
+    const bool return_inverse,
+    const bool return_counts) {
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  auto options = self.options().dtype(kLong);
+  int64_t num_inp = self.numel();
+  TORCH_CHECK(
+      num_inp <= INT_MAX,
+      "num_inp ",
+      num_inp,
+      " is too big to be handled by cub");
+  Tensor sorted;
+  if (consecutive) {
+    sorted = self;
+  } else {
+    sorted = at::empty({num_inp}, self.options());
+  }
+  scalar_t* sorted_data = sorted.data_ptr<scalar_t>();
+
+  Tensor sorted_indices;
+  if (!return_inverse) {
+    if (!consecutive) {
+      cuda::cub::sort_keys(self.data_ptr<scalar_t>(), sorted_data, num_inp);
+    }
+  } else {
+    sorted_indices = at::arange(0, num_inp, options);
```

Comment:
We no longer need this when `consecutive` is true, so I believe you can move the `arange` call inside the if clause below